src/lib/libast/features/fs

```diff
@@ -179,28 +179,39 @@ cat{
 	extern int	fstatvfs(int, struct statvfs*);
 	extern int	statvfs(const char*, struct statvfs*);
 	#endif
+
+	/*
+	 * NetBSD only provides getfsstat for binary compatibility, and
+	 * does not expose it in the system headers. Using it will
+	 * cause the build to fail with C99 because of an implicit
+	 * function error. We don't actually need this function,
+	 * so it's better to just avoid using it.
+	 */
+	#if defined(__NetBSD__)
+	#undef _lib_getfsstat
+	#endif
```

Comment:
I'd rather we just define an extern on NetBSD and any other systems where the headers don't expose it. It should work to add the following at the end of features/fs:

```
header sys/param.h
header sys/ucred.h
header sys/mount.h
extern getfsstat int (struct statfs*, int, int)
```

See features/sys and `iffe --man` for comparison and information.

Reply:
I tested this and it also fixes the build error, but causes nearly every single file compiled to complain with the following warning:
```
In file included from ./ast_fcntl.h:5:0,
                 from /home/johno/ksh/src/lib/libast/include/ast_std.h:68,
                 from /home/johno/ksh/src/lib/libast/include/ast.h:32,
                 from /home/johno/ksh/src/lib/libast/hash/hashlib.h:29,
                 from /home/johno/ksh/src/lib/libast/hash/hashlook.c:26:
./ast_fs.h:13:29: warning: 'struct statfs' declared inside parameter list will not be visible outside of this definition or declaration
 extern int getfsstat(struct statfs*, int, int);
                             ^~~~~~
```

Like getfsstat, `struct statfs` is also not readily exposed by NetBSD's headers, resulting in this warning. I personally would rather avoid fighting against the system headers. It's fairly obvious from how the headers hide this function that getfsstat shouldn't be used on NetBSD.
Granted, the current method used is lacking. Perhaps a regression test could be added to see if the function actually compiles before defining _lib_getfsstat. The getfsstat_statvfs test already does this, albiet it appears to focus moreso on if getfsstat uses statvfs than anything else.

Reply:
I've pushed a new feature getfsstat_statfs feature test and modified the old one to only define _lib_getfsstat if it actually compiles (i.e., _lib_getfsstat is no longer defined just because the function exists). As far as I can tell the compiler error is gone and there are no additional compiler warnings now either (tested on NetBSD, OpenBSD and FreeBSD).

Reply:
Thanks. You're right, that is the best solution.